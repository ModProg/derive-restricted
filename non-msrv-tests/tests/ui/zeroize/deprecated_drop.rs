use std::marker::PhantomData;

extern crate zeroize_ as zeroize;

use derive_where::derive_where;

#[derive_where(Zeroize(drop))]
struct DeprecatedDrop<T>(PhantomData<T>);

fn main() {}
