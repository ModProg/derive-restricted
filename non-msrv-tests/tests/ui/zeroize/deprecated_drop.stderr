error: `Zeroize(drop)` is deprecated, use `ZeroizeOnDrop` instead
 --> tests/ui/zeroize/deprecated_drop.rs:8:24
  |
8 | #[derive_where(Zeroize(drop))]
  |                        ^^^^
